core/bubbles/text_bubble.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Bubble} from './bubble.js';
+import {Coordinate} from '../utils/coordinate.js';
+import * as dom from '../utils/dom.js';
+import {Rect} from '../utils/rect.js';
+import {Size} from '../utils/size.js';
+import {Svg} from '../utils/svg.js';
+import {WorkspaceSvg} from '../workspace_svg.js';
+
+export class TextBubble extends Bubble {
+  private paragraph: SVGTextElement;
+
+  constructor(
+    private text: string,
+    protected readonly workspace: WorkspaceSvg,
+    protected anchor: Coordinate,
+    protected ownerRect?: Rect
+  ) {
+    super(workspace, anchor, ownerRect);
+    this.paragraph = this.stringToSvg(text, this.contentContainer);
+    this.updateBubbleSize();
+  }
+
+  /** @return the current text of this text bubble. */
+  getText(): string {
+    return this.text;
+  }
+
+  /** Sets the current text of this text bubble, and updates the display. */
+  setText(text: string) {
+    this.text = text;
+    dom.removeNode(this.paragraph);
+    this.paragraph = this.stringToSvg(text, this.contentContainer);
+    this.updateBubbleSize();
+  }
+
+  /**
+   * Converts the given string into an svg containing that string,
+   * broken up by newlines.
+   */
+  private stringToSvg(text: string, container: SVGGElement) {
```

Comment:
You're calling `stringToSvg` with the same `container` in both cases (`this.contentContainer`) so you could remove the param and just access it inside the method.

(Optional change)

Reply:
It's my preference to keep methods pure if possible. So I'm going to use as is, but if people prefer the other style we can always switch later since this is private.